x-pack/plugins/security_solution/public/detections/components/alerts_kpis/common/components.tsx

```diff
@@ -25,24 +25,41 @@ export const KpiPanel = styled(EuiPanel)<{ height?: number }>`
   }
 `;
 interface StackedBySelectProps {
-  selected: AlertsStackByField;
-  onSelect: (selected: AlertsStackByField) => void;
+  selected: string;
+  onSelect: (selected: string) => void;
 }
 
-export const StackBySelect: React.FC<StackedBySelectProps> = ({ selected, onSelect }) => {
-  const setSelectedOptionCallback = useCallback(
-    (event: React.ChangeEvent<HTMLSelectElement>) => {
-      onSelect(event.target.value as AlertsStackByField);
+export const StackByComboBoxWrapper = styled.div`
+  min-width: 350px;
+`;
+
+export const StackByComboBox: React.FC<StackedBySelectProps> = ({ selected, onSelect }) => {
+  const onChange = useCallback(
+    (options) => {
+      if (options && options.length > 0) {
+        onSelect(options[0].value);
+      } else {
+        onSelect('');
+      }
     },
     [onSelect]
   );
-
+  const selectedOptions = useMemo(() => {
+    return [{ label: selected, value: selected }];
+  }, [selected]);
+  const stackOptions = useStackByFields();
   return (
-    <EuiSelect
-      onChange={setSelectedOptionCallback}
-      options={alertsStackByOptions}
-      prepend={i18n.STACK_BY_LABEL}
-      value={selected}
-    />
+    <StackByComboBoxWrapper>
+      <EuiComboBox
+        aria-label={i18n.STACK_BY_ARIA_LABEL}
+        placeholder={i18n.STACK_BY_PLACEHOLDER}
+        prepend={i18n.STACK_BY_LABEL}
+        singleSelection={{ asPlainText: true }}
```

Comment:
Refactoring `{ asPlainText: true }` into a constant outside the component or into a ref inside the component will prevent rerenders of the `EuiComboBox`.